### ISSUE TITLE
Make `to_pil_image` do an implicit type conversion

### ISSUE THREAD

**Issue by ailzhang** [pytorch/vision#1162]
Related to https://github.com/pytorch/pytorch/issues/21044. As suggested by @ppwwyyxx , we probably should add a type conversion in https://github.com/pytorch/vision/blob/master/torchvision/transforms/functional.py#L104 when the mode is variant of `int8`. 
Right now if there're negative values in tensor, we'll just keep it as negative, which doesn't align with int8 range, and cause visual overshoot. 

Thoughts? @soumith @fmassa 
example script: 
```
import torch
from torch.nn.functional import interpolate
from PIL import Image
from torchvision.transforms.functional import to_tensor, to_pil_image
import numpy as np
import pdb

image_path = './dog.jpg'
scale_factor = 2.0

image = Image.open(image_path)

# PyTorch
image_tensor = to_tensor(image)
image_tensor.unsqueeze_(0)
image_tensor = interpolate(image_tensor, scale_factor=scale_factor, mode='bicubic', align_corners=False)
image_tensor.squeeze_(0)
# image_tensor contains negative values
to_pil_image(image_tensor, mode='RGB').save('pt.png')
```

**Comment by fmassa** [comment]
The type conversion is done in https://github.com/pytorch/vision/blob/d9830d8688bc2f2fc8070a4688d590c7c0686b27/torchvision/transforms/functional.py#L139
which makes everything be between 0-255.

I suppose you mean that we should clamp the values between 0-255 (or 0-1) before casting to bool?

This is could be a solution.

But the logic in `to_pil_image` (and `to_tensor`) is already complicated / confusing, and it might merit a refactor. I'd be ok merging a patch for a quick fix, but ideally we would have a better story around those operators. One step towards it would be to make `interpolate` (and some other methods like `grid_sample`) work on `uint8` type of data as well. But that's another story.

**Comment by ailzhang** [comment]
Yep what we need is indeed a clamp and we already do the type conversion correctly. Also agree on the logic in `to_pil_image` is already confusing, let's leave this proposal here for some time and see if this gets popular or not before we implement it.  ;)
